homeassistant/components/mopar/__init__.py

```diff
@@ -0,0 +1,110 @@
+"""Support for Mopar vehicles."""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_PIN
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['motorparts==1.1.0']
+
+DOMAIN = 'mopar'
+DATA_UPDATED = '{}_data_updated'.format(DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_VEHICLE_INDEX = 'vehicle_index'
+COOKIE_FILE = 'mopar_cookies.pickle'
+SUCCESS_RESPONSE = 'completed'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(days=7)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_PIN): cv.positive_int,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Mopar component."""
+    import motorparts
+    cookie = hass.config.path(COOKIE_FILE)
+    try:
+        session = motorparts.get_session(
+            config.get(CONF_USERNAME), config.get(CONF_PASSWORD),
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.